api/src/main/java/org/openmrs/api/db/hibernate/AuditableInterceptor.java

```diff
@@ -117,8 +118,8 @@ private boolean setCreatorAndDateCreatedIfNull(Object entity, Object[] currentSt
 	private boolean changeProperties(Object[] currentState, String[] propertyNames, boolean objectWasChanged,
 	        Map<String, Object> propertyValues, Boolean setNullOnly) {
 		
-		for (String property : propertyValues.keySet()) {
-			if (changePropertyValue(currentState, propertyNames, property, propertyValues.get(property), setNullOnly)) {
+		for (Map.Entry<String, Object> e : propertyValues.entrySet()) {
```

Comment:
@wluyima, meinsen's change is already a performance optimization. The issue is that when you use propertyValues.get(property) it may be O(log n) whereas you can simply do e.getValue() when using entrySet which is O(1).